servers/src/common/adapters.rs

```diff
@@ -361,7 +361,9 @@ impl p2p::ChainAdapter for NetToChainAdapter {
 			w(&self.chain).txhashset_write(h, txhashset_data, self.sync_state.as_ref())
 		{
 			error!(LOGGER, "Failed to save txhashset archive: {}", e);
-			!e.is_bad_data()
+			let is_good_data = !e.is_bad_data();
+			self.sync_state.set_sync_error(types::Error::Chain(e));
```

Comment:
As a less verbose alternative, you can also clone the error here. They're very small objects so it's not a big deal.

Reply:
Tried before, unfortunately `Clone is not implemented for failure::Context<error::ErrorKind>`. Have no idea about how to fix it.

And a long story about Error clone,  many Error can't clone, for some reasons such as io::Error, then all Error which packing io::Error can't clone.